numpy/lib/function_base.py

```diff
@@ -782,7 +782,7 @@ def histogram(a, bins=10, range=None, normed=False, weights=None,
         bins = bin_edges
     else:
         bins = asarray(bins)
-        if (np.diff(bins) < 0).any():
+        if np.any(bins[1:] <= bins[:-1]):
```

Comment:
This isn't equivalent - previously we allowed repeated bins, but this code does not.

I'd prefer `np.any(bins[:-1] > bins[1:])` here, which reads as `current > next`